reporter/src/main/kotlin/CopyrightStatementsProcessor.kt

```diff
@@ -0,0 +1,278 @@
+/*
+ * Copyright (c) 2017-2018 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.reporter
+
+import com.here.ort.model.yamlMapper
+import java.lang.StringBuilder
+
+private data class Parts(
+        val prefix: String,
+        val years: Set<Int>,
+        val owner: String,
+        val originalStatements: List<String>
+)
+
+private fun String.removeDuplicateWhitespaces() =
+        replace("(?=.*)([ ]+)".toRegex(), " ")
+
+private fun String.removeTrailingSpecialCharacters() =
+        replace("^([<> \\(\\)\\[\\];\\.,\\-\\+\\~&]*)".toRegex(), "")
```

Comment:
I did what you proposed which was interresting because it showed that the character sets for (1) stripping owner prefix and (2) creating normalized customer key have to be different.

I've just pushed another path where I make this code specific to 'owner' key and or  prefix removal.